src/Security/OpenIdConfigurationProvider.php

```diff
@@ -72,81 +107,135 @@ public function getResourceOwnerDetailsUrl(AccessToken $token): string
+    {
+        return $this->scopes;
+    }
 
-        if (null === $json = json_decode($content, true)) {
-            throw new \Exception('Cannot decode OpenID configuration file.');
+    /**
+     * {@inheritdoc}
+     */
+    protected function checkResponse(ResponseInterface $response, $data)
+    {
+        $error = null;
+        if (!empty($data[$this->responseError])) {
+            $error = $data[$this->responseError];
+            if (!is_string($error)) {
+                $error = var_export($error, true);
+            }
         }
-
-        if (false === file_put_contents($this->cachePath, sprintf('<?php return %s;', var_export($json, true)))) {
-            throw new \Exception('Cannot save OpenID configuration cache file.');
+        if ($error || $response->getStatusCode() >= 400) {
+            throw new IdentityProviderException($error, 0, $data);
         }
     }
 
     /**
-     * {@inheritDoc}
+     * @inheritdoc
      */
-    protected function getConfigurableOptions(): array
+    protected function createResourceOwner(array $response, AccessToken $token)
     {
-        return array_merge(parent::getConfigurableOptions(), [
-            'cacheDuration',
-        ]);
+        return new GenericResourceOwner($response, $this->responseResourceOwnerId);
     }
 
     /**
-     * {@inheritDoc}
+     * Fetch OpenID Connect Metadata
+     *
+     * @return array
+     *   Array of configuration values
+     *
+     * @throws ItkOpenIdConnectException
      */
-    protected function getRequiredOptions(): array
+    private function fetchOpenIDConnectMetadata(): array
     {
-        return [
-            'urlConfiguration',
-            'cachePath',
-        ];
-    }
+        try {
+            $response = $this->getHttpClient()->request('GET', $this->openIDConnectMetadataUrl);
+
+            if (200 !== $response->getStatusCode()) {
+                throw new ItkOpenIdConnectException('Cannot access OpenID configuration resource.');
+            }
 
+            $content = $response->getBody()->getContents();
 
+            return json_decode($content, true, 512, JSON_THROW_ON_ERROR);
+        } catch (GuzzleException $e) {
+            // @TODO 1
+        } catch (RuntimeException $e) {
+            // @TODO 2
+        } catch (JsonException $e) {
+            // @TODO 3
+        }
+    }
 
     /**
-     * Get Configuration.
+     * Get Configuration option for key.
      *
-     * @throws \Exception
+     * @param string $key
+     *  The configuration key
+     *
+     * @return string
+     *   The configuration value for the given key
      */
     private function getConfiguration(string $key): string
     {
-        $this->loadCache();
+        try {
+            $item = $this->cacheItemPool->getItem(self::CACHE_KEY);
+            if ($item->isHit()) {
+                $configuration = $item->get();
+            } else {
+                $configuration = $this->fetchOpenIDConnectMetadata();
```

Comment:
You should probably save the fetched configuration in the cache for later retrievement.

Reply:
This is fixed in https://github.com/itk-dev/openid-connect/pull/8